frontend/ui/widget/imagewidget.lua

```diff
@@ -81,12 +81,19 @@ local ImageWidget = Widget:new{
     -- If scale_for_dpi is true image will be rescaled according to screen dpi
     scale_for_dpi = false,
 
-    -- When scale_factor is not nil, native image is scaled by this factor
-    -- (if scale_factor == 1, native image size is kept)
-    -- Special case : scale_factor == 0 : image will be scaled to best fit provided
-    -- width and height, keeping aspect ratio (scale_factor will be updated
-    -- from 0 to the factor used at _render() time)
+    -- When scale_factor is not nil, native image is scaled by this factor,
+    --   (if scale_factor == 1, native image size is kept)
+    --   Special case: scale_factor == 0 : image will be scaled to best fit provided
+    --   width and height, keeping aspect ratio (scale_factor will be updated
+    --   from 0 to the factor used at _render() time)
+    -- If scale_factor is nil and stretch_limit_percantage is provided:
+    --   If the aspect ratios of the image and the widget don't differ more than stretch_limit_percentage,
+    --      then stretch
+    --      else scale
+    --   the image to best fit the container.
```

Comment:
This indentation & newline feel a bit excessive, it's just english text after all :)
`If the aspect ratios of the image and the width/height provided don't differ by more than stretch_limit_percentage, then stretch the image (as scale_factor=nil); otherwise, scale to best fit (as scale_factor=0)` (if I understand it right, from just reading that comment and not yet the code below).